### PR TITLE
Bind clone alternative

### PR DESCRIPTION
**Pull request by mattbennett**

### CODE REVIEW

File: nameko/extensions.py [hunk 1]
```diff
@@ -21,6 +21,7 @@
 
 class Extension(object):
 
+    container = None
     __clone = False
```
Comment (davidszotten) [inline nameko/extensions.py]: i guess we can drop `__clone` and have `is_clone` just look for `container`?

File: nameko/extensions.py [hunk 2]
```diff
@@ -72,19 +73,26 @@ def kill(self):
...
 
-    def clone(self, container):
-        if self.is_clone:
-            raise RuntimeError('Cloned extensions cannot be cloned.')
+    def bind(self, container):
+        """ Get an instance of this Extension to bind to `container`.
+        """
+
+        def clone(prototype):
+            if prototype.is_clone:
```
Comment (davidszotten) [inline nameko/extensions.py]: actually, we could make `clone` an internal concept and refer to them as `bound` ?

File: nameko/extensions.py (hunk 1 above)
Comment (mattbennett) [inline nameko/extensions.py]: I think so

File: nameko/extensions.py (hunk 2 above)
Comment (mattbennett) [inline nameko/extensions.py]: Yep, ``is_clone`` will become ``is_bound``. Cloning is an implementation detail.

**Pull request merged**